app/models/flood_risk_engine/address_search.rb

```diff
@@ -0,0 +1,5 @@
+module FloodRiskEngine
+  class AddressSearch < ActiveRecord::Base
```

Comment:
Saving the postcode to the database has allowed the process to work without custom controller actions, which I think is good. Personally, I'd store the postcode on the address object being built rather than creating a special class to hold the postcode during address building, but I guess this approach will allow the "address_search" step to be made more easily generic and be put in front of any address building step.

However, I think you need to clear the enrollment.address_search after the address has been built. Otherwise, the postcode from the previous address building, will appear at the start of the next address building.